pkg/api/util/conditions.go

```diff
@@ -131,11 +131,12 @@ func GetCertificateRequestCondition(req *cmapi.CertificateRequest, conditionType
 // SetCertificateCondition will set a 'condition' on the given Certificate.
 // - If no condition of the same type already exists, the condition will be
 //   inserted with the LastTransitionTime set to the current time.
-// - If a condition of the same type and state already exists, the condition
-//   will be updated but the LastTransitionTime will not be modified.
+// - If a condition of the same type and state already exists for the same
```

Comment:
I don't think it makes sense to add `for the same generation` here - the LastTransitionTime should only be updated when the state of the condition changes (e.g. from True->False or False->True) - not just if any of the fields within it change.

Also see https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions - we should be updating to align with this condition format with the rules described there :)

Reply:
So to be clear, the observed generation will be updated, but the last transition time will not be?